lib/process-services-cloud/src/lib/services/user-preference-cloud.service.ts

```diff
@@ -38,25 +38,30 @@ export class UserPreferenceCloudService extends BaseCloudService implements Pref
    * @param appName Name of the target app
    * @returns List of user preferences
    */
-  getPreferences(appName: string, key?: string): Observable<any> {
+  getPreferences(appName: string): Observable<any> {
     if (appName) {
       const url = `${this.getBasePath(appName)}/preference/v1/preferences`;
-
-      return this.get(url).pipe(map((preferences: any) => {
-        if (key) {
-          const preferencesList = preferences?.list?.entries ?? [];
-          const searchedPreferences = preferencesList.find(preference => preference.entry.key === key);
-          return searchedPreferences ? JSON.parse(searchedPreferences.entry.value) : null;
-        }
-
-        return preferences;
-      }));
+      return this.get(url);
     } else {
       this.logService.error('Appname is mandatory for querying preferences');
       return throwError('Appname not configured');
     }
   }
 
+  /**
+   * Search and return user preference in all preferences
+   *
+   * @param appName Name of the target app
+   * @returns user preferences
+   */
+  searchPreferences<T>(appName: string, key: any): Observable<T> {
+    return this.getPreferences(appName).pipe(map(preferences => {
```

Comment:
At this point this method is the new GetPreferences, but why split the two things?